winpr/libwinpr/utils/wlog/Appender.c

```diff
@@ -21,11 +21,77 @@
 #include "config.h"
 #endif
 
-#include <winpr/wlog.h>
+#include "Appender.h"
 
-#include "wlog/Layout.h"
+void WLog_Appender_Free(wLog* log, wLogAppender* appender)
+{
+	if (!appender)
+		return;
+
+	if (appender->Layout)
+	{
+		WLog_Layout_Free(log, appender->Layout);
+		appender->Layout = NULL;
+	}
 
-#include "wlog/Appender.h"
+	DeleteCriticalSection(&appender->lock);
+	appender->Free(appender);
+}
+
+wLogAppender* WLog_GetLogAppender(wLog* log)
+{
+	if (!log)
+		return NULL;
+
+	if (!log->Appender)
+		return WLog_GetLogAppender(log->Parent);
+
+	return log->Appender;
+}
+
+BOOL WLog_OpenAppender(wLog* log)
+{
+	int status = 0;
+	wLogAppender* appender;
+
+	appender = WLog_GetLogAppender(log);
+
+	if (!appender)
+		return FALSE;
+
+	if (!appender->Open)
+		return TRUE;
+
+	if (!appender->State)
+	{
+		status = appender->Open(log, appender);
+		appender->State = 1;
```

Comment:
Again with names: may I suggest Opened or Active instead of State and move to BOOL as we're using only the 0 and 1 value ?